Make pseudo attribute values of DateTimeFieldElements proper ShadowPseudoIds

annevk opened the pull request:
#### f8e7643db02e7e5c994fedf49333064c659dc2a4
<pre>
Make pseudo attribute values of DateTimeFieldElements proper ShadowPseudoIds
<a href="https://bugs.webkit.org/show_bug.cgi?id=266951">https://bugs.webkit.org/show_bug.cgi?id=266951</a>

Reviewed by Tim Nguyen.

This will enable us to have a central source of truth for
ShadowPseudoIds in due course.

Also call setPseudo() directly instead of using a somewhat useless
wrapper.

* Source/WebCore/html/shadow/DateTimeFieldElement.cpp:
(WebCore::DateTimeFieldElement::initialize): Deleted.
* Source/WebCore/html/shadow/DateTimeFieldElement.h:
* Source/WebCore/html/shadow/DateTimeFieldElements.cpp:
(WebCore::DateTimeDayFieldElement::create):
(WebCore::DateTimeHourFieldElement::create):
(WebCore::DateTimeMeridiemFieldElement::create):
(WebCore::DateTimeMillisecondFieldElement::create):
(WebCore::DateTimeMinuteFieldElement::create):
(WebCore::DateTimeMonthFieldElement::create):
(WebCore::DateTimeSecondFieldElement::create):
(WebCore::DateTimeSymbolicMonthFieldElement::create):
(WebCore::DateTimeYearFieldElement::create):
* Source/WebCore/html/shadow/DateTimeNumericFieldElement.cpp:
(WebCore::DateTimeNumericFieldElement::initialize): Deleted.
* Source/WebCore/html/shadow/DateTimeNumericFieldElement.h:
* Source/WebCore/html/shadow/DateTimeSymbolicFieldElement.cpp:
(WebCore::DateTimeSymbolicFieldElement::initialize): Deleted.
* Source/WebCore/html/shadow/DateTimeSymbolicFieldElement.h:
* Source/WebCore/html/shadow/ShadowPseudoIds.cpp:
(WebCore::ShadowPseudoIds::webkitDateAndTimeValue):
(WebCore::ShadowPseudoIds::webkitDatetimeEditDayField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditHourField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditMeridiemField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditMillisecondField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditMinuteField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditMonthField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditSecondField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditText):
(WebCore::ShadowPseudoIds::webkitDatetimeEditYearField):
* Source/WebCore/html/shadow/ShadowPseudoIds.h:

Canonical link: <a href="https://commits.webkit.org/272550@main">https://commits.webkit.org/272550@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f2626828c7db9798eabc4498d943a710c0a946c3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32065 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/10783 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/33852 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/34608 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29060 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/32894 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13140 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8006 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/28644 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/32464 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9103 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/28661 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7896 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8069 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/28576 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/35952 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29164 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29031 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34176 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8178 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6136 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32035 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/9819 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7491 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8825 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/8703 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
> Question for reviewer: why is ShadowPseudoIds::cue() not behind ENABLE(VIDEO)? Should these be behind ENABLE(DATE_AND_TIME_INPUT_TYPES)?

Yes, I think they should all be properly guarded. Can you fix this in this commit?

nt1m commented on Source/WebCore/html/shadow/ShadowPseudoIds.h:
```diff
@@ -56,6 +56,15 @@ const AtomString& webkitDatetimeEditText();
+const AtomString& webkitDatetimeEditMonthField();
+const AtomString& webkitDatetimeEditSecondField();
+const AtomString& webkitDatetimeEditYearField();
+
```
Can we group all the webkitdatetime stuff together?

annevk committed:
Make pseudo attribute values of DateTimeFieldElements proper ShadowPseudoIds
https://bugs.webkit.org/show_bug.cgi?id=266951

Reviewed by Tim Nguyen.

This will enable us to have a central source of truth for
ShadowPseudoIds in due course.

Also call setPseudo() directly instead of using a somewhat useless
wrapper.

* Source/WebCore/html/shadow/DateTimeFieldElement.cpp:
(WebCore::DateTimeFieldElement::initialize): Deleted.
* Source/WebCore/html/shadow/DateTimeFieldElement.h:
* Source/WebCore/html/shadow/DateTimeFieldElements.cpp:
(WebCore::DateTimeDayFieldElement::create):
(WebCore::DateTimeHourFieldElement::create):
(WebCore::DateTimeMeridiemFieldElement::create):
(WebCore::DateTimeMillisecondFieldElement::create):
(WebCore::DateTimeMinuteFieldElement::create):
(WebCore::DateTimeMonthFieldElement::create):
(WebCore::DateTimeSecondFieldElement::create):
(WebCore::DateTimeSymbolicMonthFieldElement::create):
(WebCore::DateTimeYearFieldElement::create):
* Source/WebCore/html/shadow/DateTimeNumericFieldElement.cpp:
(WebCore::DateTimeNumericFieldElement::initialize): Deleted.
* Source/WebCore/html/shadow/DateTimeNumericFieldElement.h:
* Source/WebCore/html/shadow/DateTimeSymbolicFieldElement.cpp:
(WebCore::DateTimeSymbolicFieldElement::initialize): Deleted.
* Source/WebCore/html/shadow/DateTimeSymbolicFieldElement.h:
* Source/WebCore/html/shadow/ShadowPseudoIds.cpp:
(WebCore::ShadowPseudoIds::webkitDateAndTimeValue):
(WebCore::ShadowPseudoIds::webkitDatetimeEditDayField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditHourField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditMeridiemField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditMillisecondField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditMinuteField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditMonthField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditSecondField):
(WebCore::ShadowPseudoIds::webkitDatetimeEditText):
(WebCore::ShadowPseudoIds::webkitDatetimeEditYearField):
* Source/WebCore/html/shadow/ShadowPseudoIds.h:

Canonical link: https://commits.webkit.org/272550@main
